fastlane/lib/fastlane/actions/download_dsyms.rb

```diff
@@ -52,12 +52,33 @@ def self.run(params)
         UI.message(message.join(" "))
 
         app.tunes_all_build_trains(platform: platform).each do |train|
+          # We try to check first with version_string, but also we try to check with display_version_string since
+          # this value could be different.
+          #
+          # Example:
+          # - App Store version set as 1.0 and app version set as 1.0.0;
+          #
+          # In that case, the response from API will be "versionString = 1.0" and "displayVersionString = 1.0.0"
           UI.verbose("Found train: #{train.version_string}, comparing to supplied version: #{version}")
-          if version && version != train.version_string
+          if version && train.version_string == train.display_version_string && version != train.version_string
             UI.verbose("Version #{version} doesn't match: #{train.version_string}")
             next
+          else
+            UI.verbose("Failed. Comparing to display_version_string: #{train.display_version_string}")
```

Comment:
@bguidolim I _think_ this `else` needs to get reworked. This `else` will get executed if `version` is `nil` which I don't think we want. I think it almost needs to get reworked something like...

```rb
if version
  if train.version_string == train.display_version_string && version != train.version_string
    UI.verbose("Version #{version} doesn't match: #{train.version_string}")
    next
  else
    UI.verbose("Failed. Comparing to display_version_string: #{train.display_version_string}")
    if version != train.display_version_string
      UI.verbose("Version #{version} doesn't match: #{train.display_version_string}")
      next
    end
  end
end
```

Thoughts?

Reply:
Yeah! Now I see it. Fixing now.